src/compiler/transformers/esnext.ts

```diff
@@ -402,6 +402,11 @@ namespace ts {
     };
 
     export function createAssignHelper(context: TransformationContext, attributesSegments: Expression[]) {
+        if (context.getCompilerOptions().target >= ScriptTarget.ES2015) {

```

Comment:
we are still checking for it under `--importHelpers` in the checker i believe. you need to change that to ES2015 as well. and also add a test for it.